src/dbnode/persist/fs/commitlog/reader.go

```diff
@@ -89,25 +79,38 @@ type reader struct {
 	infoDecoder            *msgpack.Decoder
 	infoDecoderStream      msgpack.ByteDecoderStream
 	hasBeenOpened          bool
+	fileReadID             uint64
+
+	metadataLookup map[uint64]ts.Series
```

Comment:
Do these need to be niled out when closing?

Reply:
Can do.

Reply:
Added this to close:
```go
func (r *reader) Close() error {
	err := r.chunkReader.fd.Close()
	*r = reader{}
	return err
}
```